### ISSUE TITLE
Remove a contact

### ISSUE THREAD

**Issue by tibetsprague** [holochain-apps/volla-messages#13]
AC:
- I can remove a contact from my contacts list

**Comment by MasterK0927** [comment]
@mattyg Require additional context for "remove a contact"


- Is a confirmation dialog required before removal?
- On what action remove button will be displayed?Should the "Remove" button always be visible or only on hover?

**Comment by mattyg** [comment]
>   Is a confirmation dialog required before removal?

No, lets keep it as simple as possible. No confirmation required for now.

>  On what action remove button will be displayed?Should the "Remove" button always be visible or only on hover?

Lets keep it simple as possible. For now, only display the remove button on the contact detail page.

**Comment by MasterK0927** [comment]
> > Is a confirmation dialog required before removal?
> 
> No, lets keep it as simple as possible. No confirmation required for now.
> 
> > On what action remove button will be displayed?Should the "Remove" button always be visible or only on hover?
> 
> Lets keep it simple as possible. For now, only display the remove button on the contact detail page.

@mattyg I need clarification on this. While reviewing the codebase, I observed that creating a public or private chat involves making an entry in the DHT. Therefore, I will require a backend function to handle the removal of a contact from either a public or private chat, ensuring that the corresponding entry is also removed from the DHT.

Additionally, I have the frontend flow prepared for this.

**Comment by mattyg** [comment]
@MasterK0927 Here's the backend function for deleting a contact: https://github.com/holochain-apps/volla-messages/blob/f3b99730abea2c9faf6f690ecf5d24201f1067e8/dnas/relay/zomes/coordinator/relay/src/contact.rs#L181

You should be able to follow the same pattern to call it as other calls in the RelayClient, but let me know if you want me to create the client call there as well.

**Comment by MasterK0927** [comment]
> @MasterK0927 Here's the backend function for deleting a contact:
> 
> https://github.com/holochain-apps/volla-messages/blob/f3b99730abea2c9faf6f690ecf5d24201f1067e8/dnas/relay/zomes/coordinator/relay/src/contact.rs#L181
> 
> You should be able to follow the same pattern to call it as other calls in the RelayClient, but let me know if you want me to create the client call there as well.

@mattyg  This function is for deleting the contact, not for removing the contact from the group chat only. Does we require to delete the contact entirely or just remove it from the group chat?

**Comment by mattyg** [comment]
> @mattyg This function is for deleting the contact, not for removing the contact from the group chat only. Does we require to delete the contact entirely or just remove it from the group chat?

Ah sorry yes I clarified this issue description. This issue is just for removing a contact from your contacts list, not removing it from any chats.

**Comment by MasterK0927** [comment]
> > @mattyg This function is for deleting the contact, not for removing the contact from the group chat only. Does we require to delete the contact entirely or just remove it from the group chat?
> 
> Ah sorry yes I clarified this issue description. This issue is just for removing a contact from your contacts list, not removing it from any chats.

So `"delete"` and `"remove"` holds same meaning here? If its the case, I can proceed with the frontend integration.
Thanks!

**Comment by mattyg** [comment]
> So `"delete"` and `"remove"` holds same meaning here? If its the case, I can proceed with the frontend integration. Thanks!

Correct

**Comment by MasterK0927** [comment]
@mattyg Just one more question, if we remove a contact from our contact list, will that also means to remove the user from active conversations?

**Comment by mattyg** [comment]
No for now let's just remove the contact
